notsobot/notsobot.py

```diff
@@ -718,13 +719,14 @@ def do_gascii(self, b):
             count = 0
             for frame in gif_list[:20]:
                 im = frame.copy()
-                new_im = self.generate_ascii(im)
-                img_list.append(new_im)
+                new_im, size = self.generate_ascii(im)
+                img = Image.open(new_im)
+                img_list.append(img)
                 count += 1
-            temp = BytesIO()
-            new_im.save(
-                temp, format="GIF", save_all=True, append_images=img_list, duration=0, loop=0
-            )
+                temp = BytesIO()
+                img.save(
+                    temp, format="GIF", save_all=True, append_images=img_list, duration=0, loop=0
+                )
```

Comment:
Saving inside this for-loop is a very taxing operation are you sure this is how it is intended to work?

Reply:
Ah yes, you're absolutely right. I have moved that outside of for loop. Tested my changes and it works.